pkg/resources/account_parameter.go

```diff
@@ -83,17 +83,28 @@ func DeleteAccountParameter(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*provider.Context).Client
 	key := d.Get("key").(string)
 	ctx := context.Background()
+
+	// Identify the parameter
 	parameter := sdk.AccountParameter(key)
-	defaultParameter, err := client.Parameters.ShowAccountParameter(ctx, sdk.AccountParameter(key))
+
+	// Retrieve the current default value for the parameter
+	defaultParameter, err := client.Parameters.ShowAccountParameter(ctx, parameter)
 	if err != nil {
-		return err
+		return fmt.Errorf("error retrieving default value for account parameter %s: %w", key, err)
 	}
+
 	defaultValue := defaultParameter.Default
+	if defaultValue == "" {
```

Comment:
This check is incorrect. There are parameters which defaults to none value (e.g. https://docs.snowflake.com/en/sql-reference/parameters#query-tag).